mmv1/products/firestore/api.yaml

```diff
@@ -17,9 +17,120 @@ versions:
 objects:
+  - !ruby/object:Api::Resource
+    name: 'Database'
+    min_version: beta
+    base_url: 'projects/{{project}}/databases'
+    create_url: 'projects/{{project}}/databases?databaseId={{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A Cloud Firestore Database. Currently only one database is allowed per
+      cloud project; this database must have a `database_id` of '(default)'.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/firestore/docs/'
+      api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases'
+    async: !ruby/object:Api::OpAsync
+      actions: ['create','update']
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: True
+        allowed:
+          - True
+          - False
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        required: true
+        description: |
+          Required. The ID to use for the database, which will become the final
+          component of the database's resource name. This value should be 4-63
+          characters. Valid characters are /[a-z][0-9]-/ with first character
+          a letter and the last a letter or a number. Must not be
+          UUID-like /[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}/.
+          "(default)" database id is also valid.
+        input: true
+      - !ruby/object:Api::Type::String
+        name: locationId
+        required: true
+        description: |
+          The location of the database. Available databases are listed at
+          https://cloud.google.com/firestore/docs/locations.
+        input: true
+      - !ruby/object:Api::Type::Enum
+        name: type
+        required: true
+        description: |
+          The type of the database.
+          See https://cloud.google.com/datastore/docs/firestore-or-datastore
+          for information about how to choose.
+        values:
+          - :FIRESTORE_NATIVE
+          - :DATASTORE_MODE
+      - !ruby/object:Api::Type::Enum
+        name: concurrencyMode
+        description: |
+          The concurrency control mode to use for this database.
+        values:
+          - :OPTIMISTIC
+          - :PESSIMISTIC
+          - :OPTIMISTIC_WITH_ENTITY_GROUPS
+      - !ruby/object:Api::Type::Enum
+        name: appEngineIntegrationMode
+        description: |
+          The App Engine integration mode to use for this database.
+        values:
+          - :ENABLED
+          - :DISABLED
+        default_value: :DISABLED
```

Comment:
Do we need Terraform to set the default to disabled, or could we possibly rely on the server to use a default? Reducing defaults in Terraform generally makes things easier for us to maintain and change if necessary.

Reply:
I suspect the default can be safely removed here. I will try. Thanks for the suggestion!